src/components/Carousel/index.tsx

```diff
@@ -70,23 +70,19 @@ export default class Carousel extends React.Component<CarouselProps, CarouselSta
         renderItem: (item: React.ReactNode) => {
             return item;
         },
-        renderThumbs: (children: React.ReactChild[]) => {
-            const images = Children.map<React.ReactChild | undefined, React.ReactChild>(children, (item) => {
-                let img: React.ReactChild | undefined = item;
-
-                // if the item is not an image, try to find the first image in the item's children.
-                if ((item as React.ReactElement<{ children: React.ReactChild[] }>).type !== 'img') {
-                    img = (Children.toArray((item as React.ReactElement).props.children) as React.ReactChild[]).find(
-                        (children) => (children as React.ReactElement).type === 'img'
-                    );
-                }
+        renderThumbs: (children: React.ReactNode) => {
+            const findFirstImage = (children: React.ReactNode) =>
+                Children.toArray(children)
+                    .filter(React.isValidElement)
+                    .find((child) => child.type === 'img');
 
-                if (!img) {
-                    return undefined;
-                }
+            const images = Children.toArray(children)
+                .filter((item): item is React.ReactElement => React.isValidElement(item))
```

Comment:
Here I also had to help TS by telling it that item is ReactElement, otherwise we wouldn't get the information about it having a possible children property